invenio_rdm_records/services/communities/service.py

```diff
@@ -74,21 +190,50 @@ def remove(self, identity, record_id, data, revision_id=None, uow=None):
             community_id = community["id"]
             try:
                 self._remove(identity, community_id, record)
-                uow.register(RecordCommitOp(record.parent))
-                uow.register(RecordIndexOp(record, indexer=self.indexer))
             except RecordCommunityMissing:
                 errors.append(
                     {
                         "community": community_id,
-                        "message": "The record does not belong to the community.",
+                        "message": _("The record does not belong to the community."),
                     }
                 )
             except PermissionDeniedError:
                 errors.append(
                     {
                         "community": community_id,
-                        "message": "Permission denied.",
+                        "message": _("Permission denied."),
                     }
                 )
 
+        uow.register(RecordCommitOp(record.parent))
```

Comment:
Minor: This will run despite no records were processed, maybe we could add a variable that will be set to True just after the self._remove(...) and only commit/index if some changes happen

Reply:
I will address it in the next PR: https://github.com/inveniosoftware/invenio-rdm-records/issues/1246